ouroboros-consensus/src/Ouroboros/Consensus/Storage/LedgerDB/InMemory.hs

```diff
@@ -429,26 +431,34 @@ ledgerDbPush cfg ap db =
 
 -- | Push a bunch of blocks (oldest first)
 ledgerDbPushMany :: (ApplyBlock l blk, Monad m, c)
-                 => LedgerDbCfg l
+                 => (UpdateLedgerDbTraceEvent blk -> m ())
+                 -> LedgerDbCfg l
                  -> [Ap m l blk c] -> LedgerDB l -> m (LedgerDB l)
-ledgerDbPushMany = repeatedlyM . ledgerDbPush
+ledgerDbPushMany trace = repeatedlyM . pushAndTrace
+  where
+    pushAndTrace cfg ap db = do
+      trace StartedPushingBlockToTheLedgerDb
+      res <- ledgerDbPush cfg ap db
+      trace PushedBlockToTheLedgerDb
```

Comment:
I wonder if we need this event: if we don't see the next `StartedPushingBlockToTheLedgerDb` we can assume the node is still busy validating the block logged in the current `StartedPushingBlockToTheLedgerDb` event.

Reply:
I thought that as well, but the output on logs got user confusing in the past. For example in different place there was an event `OpenedImmutableDB`, only we knew that it means that immediately we start opening `VolatileDB` after that. And if there was no feedback, they would get confused or anxious. I think tat `Starting/Done` pair of events was requested recently by Serge, right?

Reply:
answered in previous comment